dali/pipeline/operators/fused/crop_mirror_normalize.h

```diff
@@ -77,10 +201,22 @@ class CropMirrorNormalize : public Operator<Backend>, protected CropAttr {
       output_layout_ = input_layout_;
 
     CropAttr::ProcessArguments(ws);
-  }
 
-  void DataDependentSetup(Workspace<Backend> &ws);
+    const auto &in_shape = input.shape();  // This can be a copy
+    std::size_t number_of_dims = in_shape.sample_dim();
+
+    // Set internal info for each sample based on CropAttr
+    for (int data_idx = 0; data_idx < batch_size_; data_idx++) {
+      mirror_[data_idx] = spec_.template GetArgument<int>("mirror", &ws, data_idx);
+      SetupSample(data_idx, input_layout_, in_shape.tensor_shape(data_idx));
+      PrepareArgs(ws, number_of_dims, data_idx);
```

Comment:
I think you can put a loop inside `PrepareArgs` as most of the code doesn't depend there on data_idx and can be executed only once.

Reply:
Done